docs/howto/use_modules_and_layers.md

```diff
@@ -0,0 +1,392 @@
+    ZIO.accessM(_.get.getUser(userId))
+
+  def createUser(user: User): ZIO[UserRepo, DBError, Unit] =
+    ZIO.accessM(_.get.createUser(user))
+}
+```
+
+Then, we define another module to perform some basic logging. The live implementation requires a `logger: Logger`
+
+```scala mdoc:silent
+object logging {
+  trait Service {
+    def info(s: String): UIO[Unit]
+    def error(s: String): UIO[Unit]
+  }
+
+  type Logging = Has[Service]
+
+  val consoleLogger: ZLayer[Console, Nothing, Logging] = ZLayer.fromEnvironment( console =>
+    Has(
+      new Service {
+        def info(s: String): UIO[Unit] = console.get.putStrLn(s"info - $s")
+        def error(s: String): UIO[Unit] = console.get.putStrLn(s"error - $s")
+      }
+    )
+  )
+
+  //accessor methods
+  def info(s: String): ZIO[Logging, Nothing, Unit] =
+    ZIO.accessM(_.get.info(s))
+
+  def error(s: String): ZIO[Logging, Nothing, Unit] =
+    ZIO.accessM(_.get.error(s))
+}
+```
+
+The acccessor methods are provided so that we can build programs without bothering about the implementation details of the required modules, the compiler will infer fully all the required modules to complete the task
+
+```scala mdoc:silent
+val user = User(123, "Chet")
+val makeUser: ZIO[Logging with UserRepo, Nothing, Unit] = for {
+  _ <- logging.info(s"inserting user") // ZIO[Logging, Nothing, Unit]
+  _ <- userRepo.createUser(user)       // ZIO[UserRepo, DBError, Unit]
+  _ <- logging.info(s"user inserted")  // ZIO[Logging, Nothing, Unit]
+} yield ()
+```
+
+Given a program with these requirements, we can build the required layer:
+```scala
+// compose horizontally
+val horizontal: ZLayer[Console, Nothing, Logging with UserRepo] = logging.consoleLogger ++ userRepo.inMemory
+
+// fulfill missing deps, composing vertically
+val fullLayer: ZLayer.NoDeps[Nothing, Logging with UserRepo] = Console.live >>> horizontal
+
+// provide the layer to the program
+makeUser.provideLayer(fullLayer)
+  
+
+```
+
+### Example: providing partial environments
+Let's add some extra logic to our program that creates a user
+
+```scala mdoc:silent
+val makeUser: ZIO[Logging with UserRepo with Clock with Random, DBError, Unit] = for {
+    l         <- zio.random.nextLong.map(UserId)
+    createdAt <- zio.clock.currentDateTime
+    _         <- logging.info(s"inserting user")        
+    _         <- userRepo.createUser(User(l, "Chet"))   
+    _         <- logging.info(s"user inserted, created at $createdAt")
+  } yield ()
+```
+
+Now the requirements of our program are richer, and we can satisfy the partially by providing our custom layers, and leaving out the layers that are covered by the standard environment `ZEnv`, in one line of code
+
+```scala mdoc:silent
+  val zEnvMakeUser: ZIO[ZEnv, DBError, Unit] = makeUser2.provideCustomLayer(fullLayer)
+```
+
+Notice that `provideCustomLayer` is just a special case of `provideSomeLayer`.
+
+## APPENDIX: The _classic_ module formulation (until version 1.0.0-RC17)
```

Comment:
I'd completely remove the description of the old way. It's being deprecated for good reasons and we don't want to promote that. If any current early adopters has a problem switching we've got quite active discord channel.

Reply:
👍  Will do, maybe I will put this in a blog post